frontend/catalyst/debug/instruments.py

```diff
@@ -32,14 +32,50 @@
 ## API ##
 @contextmanager
 def instrumentation(session_name, filename=None, detailed=False):
-    """Start an instrumentation session. A session cannot be tied to the creation of a QJIT object
+    """Instrumentation session to output information on wall time, CPU time,
+    and intermediate program size of a program during compilation and execution.
+
+    A session cannot be tied to the creation of a QJIT object
     nor its lifetime, because it involves both compile time and runtime measurements. It cannot
     be a context-free process either since we need to write results to an existing results file.
 
     Args:
         session_name (str): identifier to distinguish multiple results, primarily for humans
-        filename (str): desired path to write results to in YAML format
-        detailed (bool): whether to instrument finegrained steps in the compiler and runtime
+        filename (str): Desired path to write results to in YAML format. If ``None``, the
+            results will instead be printed to the console.
+        detailed (bool): whether to instrument finegrained steps in the compiler and runtime.
+            If ``False``, only high-level steps such as program capture and
+            compilation are reported.
+
+    **Example**
+
+    >>> @qjit
+    ... def expensive_function(a, b):
+    ...     return a + b
+    >>> with debug.instrumentation("session_name", detailed=False):
+    >>>     expensive_function(1, 2)
+    2024-04-29 18:19:29.349886:
+    name: session_name
+    system:
+    os: Linux-6.1.58+-x86_64-with-glibc2.35
+    arch: x86_64
+    python: 3.10.12
```

Comment:
Note this is not the output of the given example (which doesn't specify the `filename` parameter). The console format is a much simpler and looks more like "logging" output rather than the YAML format shown here.

Reply:
Oops, let me fix that.